stuffer/s2n_stuffer.c

```diff
@@ -276,8 +276,10 @@ int s2n_stuffer_write_bytes(struct s2n_stuffer *stuffer, const uint8_t * data, c
     return S2N_SUCCESS;
 }
 
-int s2n_stuffer_writev_bytes(struct s2n_stuffer *stuffer, const struct iovec* iov, int iov_count, size_t offs, size_t size)
+int s2n_stuffer_writev_bytes(struct s2n_stuffer *stuffer, const struct iovec* iov, size_t iov_count, uint32_t offs, uint32_t size)
 {
+    PRECONDITION_POSIX(s2n_stuffer_is_valid(stuffer));
+    notnull_check(iov);
```

Comment:
would a `iovec_is_valid()` function make sense?

Reply:
@danielsn I don't think it'd make much sense here. This is an array of [struct iovec](https://www.gnu.org/software/libc/manual/html_node/Scatter_002dGather.html) and the only "property" we need to guarantee here is nullness. Perhaps we could add `S2N_MEM_IS_WRITABLE(iov_base, iov_len)`, but that would go for every element in the array, which will make the check expensive.